stdlib/Statistics/src/Statistics.jl

```diff
@@ -148,7 +149,8 @@ var(iterable; corrected::Bool=true, mean=nothing) = _var(iterable, corrected, me
 function _var(iterable, corrected::Bool, mean)
     y = iterate(iterable)
     if y === nothing
-        throw(ArgumentError("variance of empty collection undefined: $(repr(iterable))"))
+        T = eltype(iterable)
+        return typeof((abs2(zero(T)) + abs2(zero(T)))/2)(NaN)
```

Comment:
I would leave it as is, because this is how `var` for `AbstractArray` is implemented and we want to be non-breaking (so it should have 100% the same behavior - the same outputs and the same errors - so using array or iterator is transparent).

For sure in Julia 2.0 approach we could consider cleaning it up in both places.

Reply:
I just mean that instead of letting `zero` throw the error, it would be clearer to check `T === Any` and throw a more explicit error.

Reply:
OK. We could still print a nice error in the most common case, but then we'd lose consistency.

Reply:
This is the point for now. As discussed in Julia 2.0 we can probably clean it up in both methods.

Reply:
personally, would use `oftype` as @nalimilan suggested for clarity